clients/da-vinci-client/src/main/java/com/linkedin/davinci/blobtransfer/client/P2PFileTransferClientHandler.java

```diff
@@ -166,15 +240,15 @@ private String getFileNameFromHeader(HttpResponse response) {
 
   private void handleEndOfTransfer(ChannelHandlerContext ctx) {
     LOGGER.info("All files received successfully for {}", payload.getFullResourceName());
-    // In the short term, we decided to let the netty client handle writing files to the disk, so
-    // the future completes with a null. It's subject to change.
-    inputStreamFuture.toCompletableFuture().complete(null);
+    metadataFuture.toCompletableFuture().complete(metadata);
     ctx.close();
   }
 
   private void resetState() {
     outputFileChannel = null;
     fileName = null;
     fileContentLength = 0;
+    currentTransferType = null;
+    accumulatedMetadataContent.clear();
```

Comment:
`clear()` on a ByteBuf does not release the underlying memory; it only resets the read and write indices, making the buffer appear empty without deallocating memory. If we need to release, we need to call `release`